### ISSUE TITLE
1.38: build fails

### ISSUE THREAD

**Issue by kloczek** [geany/geany-plugins#1175]
Looks like last release no longer build
```console
make[3]: Entering directory '/home/tkloczko/rpmbuild/BUILD/geany-plugins-1.38/git-changebar/src'
/bin/sh ../../libtool  --tag=CC   --mode=compile /usr/bin/gcc -DHAVE_CONFIG_H -I. -I../..   -DG_LOG_DOMAIN=\"GitChangeBar\"  -DLOCALEDIR=\""/usr/share/locale"\" -DPREFIX=\""/usr"\" -DDOCDIR=\""/usr/share/doc/geany-plugins"\" -DGEANYPLUGINS_DATADIR=\""/usr/share"\" -DPKGDATADIR=\""/usr/share/geany-plugins"\" -DLIBDIR=\""/usr/lib64"\" -DPKGLIBDIR=\""/usr/lib64/geany-plugins"\" -DPLUGINDATADIR=\""/usr/share/geany-plugins/git-changebar"\" -DPLUGINDOCDIR=\""/usr/share/doc/geany-plugins/git-changebar"\" -DPLUGINLIBDIR=\""/usr/lib64/geany-plugins/git-changebar"\" -DPLUGIN="\"git-changebar\"" -DGTK -I/usr/include/geany -I/usr/include/geany/tagmanager -I/usr/include/geany/scintilla -I/usr/include/gtk-3.0 -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/harfbuzz -I/usr/include/cairo -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/atk-1.0  -Wall -Wimplicit-function-declaration -Wmissing-parameter-type -Wold-style-declaration -Wpointer-arith -Wshadow -Wundef -Wwrite-strings -I/usr/include/gtk-3.0 -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/harfbuzz -I/usr/include/cairo -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/atk-1.0  -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -c -o git_changebar_la-gcb-plugin.lo `test -f 'gcb-plugin.c' || echo './'`gcb-plugin.c
libtool: compile:  /usr/bin/gcc -DHAVE_CONFIG_H -I. -I../.. -DG_LOG_DOMAIN=\"GitChangeBar\" -DLOCALEDIR=\"/usr/share/locale\" -DPREFIX=\"/usr\" -DDOCDIR=\"/usr/share/doc/geany-plugins\" -DGEANYPLUGINS_DATADIR=\"/usr/share\" -DPKGDATADIR=\"/usr/share/geany-plugins\" -DLIBDIR=\"/usr/lib64\" -DPKGLIBDIR=\"/usr/lib64/geany-plugins\" -DPLUGINDATADIR=\"/usr/share/geany-plugins/git-changebar\" -DPLUGINDOCDIR=\"/usr/share/doc/geany-plugins/git-changebar\" -DPLUGINLIBDIR=\"/usr/lib64/geany-plugins/git-changebar\" -DPLUGIN=\"git-changebar\" -DGTK -I/usr/include/geany -I/usr/include/geany/tagmanager -I/usr/include/geany/scintilla -I/usr/include/gtk-3.0 -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/harfbuzz -I/usr/include/cairo -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/atk-1.0 -Wall -Wimplicit-function-declaration -Wmissing-parameter-type -Wold-style-declaration -Wpointer-arith -Wshadow -Wundef -Wwrite-strings -I/usr/include/gtk-3.0 -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/harfbuzz -I/usr/include/cairo -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/atk-1.0 -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -c gcb-plugin.c  -fPIC -DPIC -o .libs/git_changebar_la-gcb-plugin.o
gcb-plugin.c: In function 'gcb_git_buf_grow':
gcb-plugin.c:219:12: error: 'git_buf' has no member named 'asize'; did you mean 'size'?
  219 |   if (buf->asize == 0) {
      |            ^~~~~
      |            size
gcb-plugin.c: In function 'buf_zero':
gcb-plugin.c:237:10: error: 'git_buf' has no member named 'asize'; did you mean 'size'?
  237 |     buf->asize = 0;
      |          ^~~~~
      |          size
gcb-plugin.c: In function 'on_plugin_configure_response':
gcb-plugin.c:1606:7: warning: 'gtk_color_button_get_color' is deprecated: Use 'gtk_color_chooser_get_rgba' instead [-Wdeprecated-declarations]
 1606 |       gtk_color_button_get_color (GTK_COLOR_BUTTON (cw->added_color_button),
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:74,
                 from gcb-plugin.c:27:
/usr/include/gtk-3.0/gtk/gtkcolorbutton.h:92:12: note: declared here
   92 | void       gtk_color_button_get_color      (GtkColorButton *button,
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~
gcb-plugin.c:1609:7: warning: 'gtk_color_button_get_color' is deprecated: Use 'gtk_color_chooser_get_rgba' instead [-Wdeprecated-declarations]
 1609 |       gtk_color_button_get_color (GTK_COLOR_BUTTON (cw->changed_color_button),
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/gtk-3.0/gtk/gtkcolorbutton.h:92:12: note: declared here
   92 | void       gtk_color_button_get_color      (GtkColorButton *button,
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~
gcb-plugin.c:1612:7: warning: 'gtk_color_button_get_color' is deprecated: Use 'gtk_color_chooser_get_rgba' instead [-Wdeprecated-declarations]
 1612 |       gtk_color_button_get_color (GTK_COLOR_BUTTON (cw->removed_color_button),
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/gtk-3.0/gtk/gtkcolorbutton.h:92:12: note: declared here
   92 | void       gtk_color_button_get_color      (GtkColorButton *button,
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~
gcb-plugin.c: In function 'plugin_configure':
gcb-plugin.c:1680:5: warning: 'gtk_color_button_set_color' is deprecated: Use 'gtk_color_chooser_set_rgba' instead [-Wdeprecated-declarations]
 1680 |     gtk_color_button_set_color (GTK_COLOR_BUTTON (cw->added_color_button),
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/gtk-3.0/gtk/gtkcolorbutton.h:89:12: note: declared here
   89 | void       gtk_color_button_set_color      (GtkColorButton *button,
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~
gcb-plugin.c:1683:5: warning: 'gtk_color_button_set_color' is deprecated: Use 'gtk_color_chooser_set_rgba' instead [-Wdeprecated-declarations]
 1683 |     gtk_color_button_set_color (GTK_COLOR_BUTTON (cw->changed_color_button),
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/gtk-3.0/gtk/gtkcolorbutton.h:89:12: note: declared here
   89 | void       gtk_color_button_set_color      (GtkColorButton *button,
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~
gcb-plugin.c:1686:5: warning: 'gtk_color_button_set_color' is deprecated: Use 'gtk_color_chooser_set_rgba' instead [-Wdeprecated-declarations]
 1686 |     gtk_color_button_set_color (GTK_COLOR_BUTTON (cw->removed_color_button),
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/gtk-3.0/gtk/gtkcolorbutton.h:89:12: note: declared here
   89 | void       gtk_color_button_set_color      (GtkColorButton *button,
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by kloczek** [comment]
And just tested 1.38 + master commits and build fails the same way 😞

**Comment by elextr** [comment]
See #1165 for attempt to fix that was aborted due to lack of sufficient knowledge of how to fix.  Input from libgit experts is welcome.

**Comment by kloczek** [comment]
> See #1165 for attempt to fix that was aborted due to lack of sufficient knowledge of how to fix. Input from libgit experts is welcome.

With that PR applied (cleanly) on top of master I was able build geany-plugins (with passing test suiite)
Thank you 👍 
```console
Wrote: /home/tkloczko/rpmbuild/SRPMS/geany-plugins-1.38-5.fc35.src.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geanynumberedbookmarks-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geanyctags-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geanymacro-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geniuspaste-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-commander-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geanydoc-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-codenav-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geniuspaste-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-spellcheck-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-git-changebar-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geanymacro-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-spellcheck-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-git-changebar-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-pohelper-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geanyctags-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-pretty-printer-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-autoclose-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-overview-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-treebrowser-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geanyprj-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-pohelper-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-workbench-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-projectorganizer-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geanyvc-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geanyprj-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-addons-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-latex-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geanypg-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geanygendoc-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-vimode-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-vimode-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-debugger-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-treebrowser-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-projectorganizer-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geanyvc-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geanygendoc-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-workbench-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-overview-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-addons-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-debugger-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-markdown-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-markdown-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-scope-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geanyinsertnum-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-pretty-printer-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-xmlsnippets-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-latex-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-defineformat-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geanyminiscript-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-scope-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geanynumberedbookmarks-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-lipsum-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-defineformat-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-automark-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geanypg-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geanyextrasel-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-tableconvert-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geanyinsertnum-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-lineoperations-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-sendmail-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-lineoperations-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-commander-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-shiftcolumn-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-updatechecker-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-autoclose-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-tableconvert-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-pairtaghighlighter-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-lipsum-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-keyrecord-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geanydoc-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-keyrecord-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-updatechecker-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-sendmail-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geanyextrasel-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-codenav-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-pairtaghighlighter-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-geanyminiscript-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-automark-debuginfo-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-xmlsnippets-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-shiftcolumn-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-1.38-5.fc35.x86_64.rpm
Wrote: /home/tkloczko/rpmbuild/RPMS/geany-plugins-debugsource-1.38-5.fc35.x86_64.rpm
```

> Geany defaults to deprecations off due to GTK3 deprecations spamming the whole build, they are irrelevant until Geany moves to GTK4.

Briefly looking on warnings I had impression that et least 20% of those warnings are relatedo to glib.

**Comment by eht16** [comment]
@kloczek 
The previously mentioned patch should not be used any longer.

A proper fix was merged into master from #1178.
